lib/print-config.js

```diff
@@ -0,0 +1,289 @@
+const parser = require('yargs-parser')
+
+/**
+ * Function: printConfig
+ *
+ * @param {Object} yargs: instance of populated yargs object.
+ * @param {Function} hideInstrumenteeArgs: Callback defined in lib/parse-args.
+ * @returns {undefined}
+ *
+ * Entry point for print config logic from lib/parse-args.js file.
+ * Kills process at the end of execution.
+ *
+ */
+function printConfig (yargs, hideInstrumenteeArgs) {
+  const argv = process.argv.slice(2)
+  const checkArgs = parser(argv)
+
+  let shouldPrint = false
+
+  if (Object.keys(checkArgs).includes('print-config')) {
+    // checkArgs['print-config'] could contain a boolean or a string
+    // representing a boolean.
+    if (typeof checkArgs['print-config'] === 'boolean') {
+      shouldPrint = checkArgs['print-config']
+    } else if (typeof checkArgs['print-config'] === 'string') {
+      shouldPrint = JSON.parse(checkArgs['print-config'])
+    }
+  }
+
+  if (shouldPrint) {
+    const args = yargs.parse(hideInstrumenteeArgs())
+    const cmdExecuted = 'c8 ' + argv.join(' ')
+    const cleanArgs = cleanUpArgumentObject(args)
+
+    if (args.printConfigFormat === 'text') {
+      printConfigText(cleanArgs, cmdExecuted)
+    } else if (checkArgs.printConfigFormat === 'json') {
+      const jsonYargs = JSON.stringify(cleanArgs, 2)
+      console.log(jsonYargs)
+    }
+
+    process.exit()
+  }
+}
+
+/**
+ * Function: cleanUpArgumentObject
+ *
+ * @param {Object} args: key/value pairs of configuration options
+ *  generated by yargs.parse().
+ * @returns {Object} - Clone of args with duplicated data removed.
+ *
+ * This function exclude duplicate values that have different keys.
+ * Additionally, scrubs convenience key values.
+ *
+ * For example: args['temp-directory'] and args['tempDirectory']
+ * are essentially the same variable.
+ *
+ */
+function cleanUpArgumentObject (args) {
+  const argsToPrint = {}
+
+  const keysToIterate = Object.keys(args).filter(v => {
+    return (!['_', '$0'].includes(v) && v.length > 1)
+  })
+
+  const camelCaseKeys = keysToIterate.filter(x => {
+    return [...x.matchAll(/([A-Z])/g)].length > 0
+  })
+
+  keysToIterate.forEach(v => {
+    if (camelCaseKeys.includes(v)) {
+      // Derive Kebab Case string from Camel Case Key string
+      const newKey = v.replace(/([A-Z])/g, '-$1').toLowerCase()
+      // If the Kebab Case key is not assigned a value
+      if (!args[newKey]) {
+        // Then assigned it the Camel Case Variable
+        argsToPrint[newKey] = args[v]
+      }
+    } else {
```

Comment:
refactor: move to one if statement

Reply:
might need to keep it as it.  There is less branching logic in the current implementation.